src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -713,12 +713,21 @@ namespace winrt::TerminalApp::implementation
             // Switch to the BG thread -
             co_await winrt::resume_background();
 
-            // Then enqueue the rest of this function for after the UI thread settles.
-            co_await wil::resume_foreground(Dispatcher(), CoreDispatcherPriority::Low);
+            if (auto self{ weak.get() })
+            {
+                // Then enqueue the rest of this function for after the UI thread settles.
+                co_await wil::resume_foreground(self->Dispatcher(), CoreDispatcherPriority::Low);
```

Comment:
I'd suggest just storing the dispatcher in the beginning as
```cpp
const auto dispatcher = Dispatcher();
```

Seems like a more hassle free solution to me.
Also: Do you mind removing the `winrt::apartment_context` instantiation?

Finally there's the option to drop all of the coroutine stuff and write this:
```cpp
Dispatcher().RunAsync(CoreDispatcherPriority::Low, [weak = get_weak()]() {
    if (auto self = weak.get())
    {
        self->_InitializedHandlers(*self, nullptr);
    }
});
```